classes/check/metriccheck.php

```diff
@@ -0,0 +1,81 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_cloudmetrics\check;
+
+use core\check\check;
+use core\check\result;
+use tool_cloudmetrics\metric\metric_item;
+
+/**
+ * Performance check for metrics
+ *
+ * @package tool_cloudmetrics
+ * @author Peter Sistrom (petersistrom@catalyst-au.net)
+ * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @copyright Catalyst IT
+ */
+class metriccheck extends check {
+
+    /** @var $metric metric_item to be checked*/
+    private $metric;
+
+    /**
+     * Constructor
+     *
+     * @param metric_item $metric
+     */
+    public function __construct($metric) {
+        $this->metric = $metric;
+        $this->id = $metric->get_name();
```

Comment:
all checks should have a unique id, at the moment it creates multiple checks with the same id:

tool_cloudmetrics_metriccheck

Reply:
Updated to create a unique id for each check